### DISCUSSION TITLE
TreeView Items property is bound to ViewModel property allItems, allItems change does not cause TreeView to update

### DISCUSSION THREAD

**Discussion by pzs-scnu** [AvaloniaUI/Avalonia#14133] [Q&A] [answered]
I have a TreeView control which its Items property is bound to my ViewModel's property:
```
<TreeView Margin="0" Items="{Binding allItems, Mode=TwoWay}" Name="albums" >
```
In my ViewModel, allItems is defined as:
```
        private List<AlbumItem> _allItems; 
        public List<AlbumItem> allItems
        {
            get => _allItems; 
            set => this.RaiseAndSetIfChanged(ref _allItems, value);
        }
```
when I use Avalonia package 11.0.0-preview5, ViewModel's allItems change can cause TreeView updates its UI, but when I upgrade Avalonia to 11.0.5 or 11.0.6, allItems property change does not cause TreeView to update its UI, my code remains unchanged, what's problem?

**Comment by timunie** [comment, marked as answer]
You need a public ObservableCollection which has a getter at least. It should also be OneWay only. This will notify the UI if items.were added or removed
